tools/bench-libs/src/main/java/org/enso/benchmarks/libs/SpecCollector.java

```diff
@@ -1,56 +1,90 @@
 package org.enso.benchmarks.libs;
 
+import java.io.ByteArrayOutputStream;
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Collection;
 import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import org.enso.polyglot.MethodNames.Module;
 import org.graalvm.polyglot.Context;
 import org.graalvm.polyglot.PolyglotException;
 import org.graalvm.polyglot.Source;
 import org.graalvm.polyglot.Value;
+import org.graalvm.polyglot.io.IOAccess;
 
 /**
  * Collect benchmark specifications from source files.
  */
 public class SpecCollector {
+  private final File rootDir;
   private final Context ctx;
+  private static final String allSuitesVarName = "all";
 
-  public SpecCollector(Context ctx) {
-    this.ctx = ctx;
+  public SpecCollector(File benchProjectDir, File languageHomeOverride) {
+    if (!benchProjectDir.exists() || !benchProjectDir.isDirectory()) {
+      throw new IllegalArgumentException(benchProjectDir + " is not a directory.");
+    }
+    this.rootDir = benchProjectDir;
+    this.ctx = Context.newBuilder()
+        .allowExperimentalOptions(true)
+        .allowIO(IOAccess.ALL)
+        .allowAllAccess(true)
+        .logHandler(new ByteArrayOutputStream())
+        .option(
+            "enso.projectRoot",
+            benchProjectDir.getAbsolutePath()
+        )
+        .option(
+            "enso.languageHomeOverride",
+            languageHomeOverride.getAbsolutePath()
+        )
+        .build();
   }
 
-  public Collection<BenchSuite> collectBenchSpecsFromDir(File rootDir) {
-    throw new UnsupportedOperationException("unimplemented");
+  public Collection<BenchSuite> collectAllBenchSpecs() {
+    try (Stream<Path> stream = Files.walk(rootDir.toPath())) {
+      return stream
+          .map(Path::toFile)
+          .filter(file -> file.isFile() && file.canRead() && file.getName().endsWith(".enso"))
```

Comment:
We are walking all the files and calling `all` in each of it. That's possible approach.

However currently the test harness as well as benchmark harness define `Main.enso` and list all the tests and benchmarks from there. If we stick to this approach, we wouldn't need to scan all the file. We'd just executed single, fixed entry point like `Main.all_benchmarks` and got them all.

I have no preference myself, however. CCing @jdunkerley and @radeusgd.

Reply:
This behavior is already changed. No more scanning all the source files. Explained in https://github.com/enso-org/enso/pull/7101/files/8f3d761e697af6ddb5e43e63f48aa899ab426a66#r1281685901